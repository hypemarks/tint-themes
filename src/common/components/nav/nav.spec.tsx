import { render } from '@testing-library/react';
import React, { ReactElement } from 'react';
import Nav from './nav';

describe('Test <Nav /> Component', () => {
  const socialFeeds = { twitter: [] };
  const navConfig = {
    publicPost: true,
    waterfallToggle: false,
    handleOnNavItemClick: jest.fn(),
    navItemColor: 'red',
    handleOnNavSearch: jest.fn(),
    navSearchColor: 'blue',
  };
  const addPostConfig = {
    text: 'text',
    style: {},
    onClick: jest.fn(),
    handleEnterKey: jest.fn(),
  };

  const ReactElement = (): ReactElement => {
    return <></>;
  };
  it('should match snapshot', () => {
    const wrapper = render(
      <Nav
        socialFeeds={socialFeeds}
        navConfig={navConfig}
        addPostConfig={addPostConfig}
        renderAddPost={ReactElement}
        renderNavSearch={ReactElement}
        renderNavList={ReactElement}
      />
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
