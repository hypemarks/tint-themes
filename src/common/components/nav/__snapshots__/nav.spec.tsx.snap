// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Nav /> Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav"
    style="margin-top: 5px;"
  >
    <div
      class="nav__add-post"
    />
    <div
      class="nav__list"
    />
  </div>
</DocumentFragment>
`;
